metaci/testresults/robot_importer.py

```diff
@@ -1,85 +1,99 @@
+    @param1 (str) The filepath to the robot results
+    """
+    test_results_path = Path(test_results_path)
+    if not test_results_path.is_file():
+        raise BuildError(
+            f"Given Robot test result file is not a file: {test_results_path}"
+        )
+
     # import is here to avoid import cycle
     from metaci.build.models import BuildFlowAsset
 
-    path = pathlib.Path(path)
-
-    # find the robot task that generated the log file so
-    # that we can access the task options when rendering
-    # the log file
-    mtime = timezone.make_aware(
-        datetime.fromtimestamp(path.stat().st_mtime), timezone.get_current_timezone()
-    )
-    robot_task = None
-    for task in build_flow.tasks.order_by("-time_end"):
-        if task.time_start <= mtime <= task.time_end:
-            robot_task = task
-            break
-
-    classes = {}
-    methods = {}
-
-    with open(path, "rb") as f:
+    with open(test_results_path, "rb") as f:
         asset = BuildFlowAsset(
-            build_flow=build_flow,
+            build_flow=flowtask.build_flow,
             asset=ContentFile(f.read(), "output.xml"),
             category="robot-output",
         )
         asset.save()
 
+    classes = {}
+    methods = {}
     suite_screenshots = {}
-    for result in parse_robot_output(path):
-        class_and_method = f"{result['suite']['name']}.{result['name']}"
-
+    for result in parse_robot_output(test_results_path):
         testclass = classes.get(result["suite"]["name"], None)
         if not testclass:
             testclass, created = TestClass.objects.get_or_create(
                 name=result["suite"]["name"],
-                repo=build_flow.build.repo,
+                repo=flowtask.build_flow.build.repo,
                 test_type="Robot",
             )
             classes[result["suite"]["name"]] = testclass
 
-        # Attach suite screenshots to buildflow
-        dirname = os.path.dirname(path)
-        for screenshot in result["suite"]["screenshots"]:
+        # Create screenshot assets for corresponding BuildFlow
+        # These screenshots are generated during robot test suite setup/teardown
+        dirname = test_results_path.parent
+        for i, screenshot in enumerate(result["suite"]["screenshots"]):
+
             if screenshot in suite_screenshots:
                 continue
-            screenshot_path = screenshot
+
             if dirname:
-                screenshot_path = f"{dirname}/{screenshot}"
+                screenshot_path = Path(f"{dirname}/{screenshot}")
+            else:
+                screenshot_path = Path(screenshot)
+
             with open(screenshot_path, "rb") as f:
                 asset = BuildFlowAsset(
-                    build_flow=build_flow,
+                    build_flow=flowtask.build_flow,
                     asset=ContentFile(f.read(), screenshot),
-                    category="robot-screenshot",
+                    category=f"robot-screenshot-{i+1}",
```

Comment:
They can and should all go in the same category. It's the name passed to ContentFile that needs to be unique. We should also incorporate the step_num into that name.